### PR TITLE
Let users set the common marker extensions

### PR DESCRIPTION
**Pull request by gwincr11**
Changes:
  - Expose the common marker extensions to the context.
  - Set a default list of ext.
  - Update docs.

### CODE REVIEW

**Comment by gjtorikian** [comment]
I'm really not sure how CI failed to install any of the gems. @jch, have you seen this before?

File: lib/html/pipeline/markdown_filter.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ class Pipeline
     #
     # Context options:
     #   :gfm      => false    Disable GFM line-end processing
+    #   :commonmarker_extensions => [ ] Common marker extensions to
```
Comment (kivikakk) [inline lib/html/pipeline/markdown_filter.rb]: What's the `[ ]` doing here? I think this is meant to suggest the default, which is instead written on the next line.

Comment (kivikakk) [inline lib/html/pipeline/markdown_filter.rb]: `Common marker` => `CommonMarker`

File: test/html/pipeline/markdown_filter_test.rb [hunk 2]
```diff
@@ -48,6 +48,20 @@ def test_fenced_code_blocks_with_language
...
   end
+
+  def test_standard_extensions
+    iframe = "<iframe src='http://www.google.com'></iframe>"
+    iframe_escaped = "&lt;iframe src='http://www.google.com'>&lt;/iframe>"
+    doc = MarkdownFilter.new(iframe).call
+    assert_equal(doc, iframe_escaped)
+  end
+
+
```
Comment (kivikakk) [inline test/html/pipeline/markdown_filter_test.rb]: One too many blank lines.

**Comment by kivikakk** [comment]
Wow, the failures sure are extensive. I'm guessing a recent merge updated `Gemfile` without the lockfiles getting updated too? Though it looks like you caught that here: https://github.com/jch/html-pipeline/commit/2852874da3f1ed5d0627dc248750b6c23c01ed72 Hrm.

**Comment by kivikakk** [comment]
➡️  https://github.com/jch/html-pipeline/pull/280

**Comment by gjtorikian** [comment]
Can you pull in `master` when you get a chance @gwincr11 ? 👍

**Comment by thejandroman** [comment]
@kivikakk @gjtorikian This should be ready for a re-review.

**Comment by gjtorikian** [comment]
Looks good to me!

**Pull request merged**